src/irmin-mem/irmin_mem.mli

```diff
@@ -33,5 +33,8 @@ module Atomic_write : Irmin.ATOMIC_WRITE_STORE_MAKER
 module Make : Irmin.S_MAKER
 (** An in-memory Irmin store. *)
 
+module Make_layered : Irmin.L_MAKER
```

Comment:
As discussed in person, it might be nice to avoid the proliferation of `Make_*` functors by expressing this as a functor over stores. I think it'll actually need to be a functor over the _existing_ `Make` functors (i.e. ∀ `Make` → `Make_layered`), to avoid the input store being too abstract to do anything useful with. Have you had a look at this?